From d776907b5a64de2f5e3cfd174519f0ff6ee82f13 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 16 Oct 2020 09:14:05 +0300
Subject: [PATCH] [SW-22029] pcie: update pcie synopsys workaround 01038891

As part of goya2 ASIC bringup, some extra register settings were
needed to be added for PCIe synopsys workaround case 01038891
to switch between gen3 and gen4 link speeds. This changes are
already made in ppboot which takes care of PCIe link bring up
and is recommended by HW team to also include in PCIe recovery
flow in arcboot ROM where synopsys workaround is already implemented.
This patch adds those additional register settings.

Change-Id: Id328f030c9c7266fb8e9d8b8f51b895034bfe103
---

diff --git a/common/pcie.c b/common/pcie.c
index 7daedf1..14e38ac 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -100,6 +100,11 @@
 			mmPCIE_PHY_PROTOCOL0_EXT_TX_EQ_POST_G1,
 			mmPCIE_PHY_PROTOCOL0_EXT_TX_EQ_POST_G4, 0);
 
+		regmap_writel(PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3_CONF,
+				mmPCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3);
+		regmap_writel(PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4_CONF,
+				mmPCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4);
+
 		regmap_writel(1, mmPCIE_PHY_PHY_EXT_CTRL_SEL);
 	}
 }
diff --git a/common/pcie.h b/common/pcie.h
index b1a6c33..040176e 100644
--- a/common/pcie.h
+++ b/common/pcie.h
@@ -18,6 +18,8 @@
 #define IATU_LWR_TARGET_ADDR_OFF_INBOUND	(IATU_BASE + 0x114)
 #define IATU_UPPER_TARGET_ADDR_OFF_INBOUND	(IATU_BASE + 0x118)
 
+#define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3_CONF	0x5555
+#define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4_CONF	0xFF
 
 void pcie_recovery(void);
 
